subprojects/core/src/main/java/org/gradle/api/internal/tasks/properties/PropertyValidationAccess.java

```diff
@@ -110,16 +104,21 @@ private static String propertyValidationMessage(Class<?> task, String qualifiedP
         return String.format("Task type '%s': property '%s' %s.", task.getName(), qualifiedPropertyName, validationMessage);
     }
 
-    private static class TypeNode extends AbstractBeanNode {
+    private static class TypeNode extends AbstractBeanNode<TypeNode> {
         private final TypeToken<?> beanType;
 
         public TypeNode(@Nullable String parentPropertyName, TypeToken<?> beanType) {
             super(parentPropertyName, beanType.getRawType());
             this.beanType = beanType;
         }
 
-        public TypeToken<?> getBeanType() {
-            return beanType;
+        @Override
+        public Iterable<TypeNode> asIterable(NestedBeanContext<TypeNode> context) {
+            @SuppressWarnings("unchecked")
+            TypeToken<Iterable> typeToken = (TypeToken<Iterable>) beanType;
```

Comment:
Use `Cast.uncheckedCast()` instead of a `@SuppressWarnings` annotation.

Reply:
Done.